tests/functional/ipset_controller_test.go

```diff
@@ -312,6 +314,72 @@ var _ = Describe("IPSet controller", func() {
 		})
 	})
 
+	When("an IPSet with multiple networks gets created", func() {
+		var netSpecs []networkv1.Network
+		var networks []networkv1.IPSetNetwork
+		BeforeEach(func() {
+			netSpecs = []networkv1.Network{}
+			netSpecs = append(netSpecs, GetNetSpec(net1, GetSubnet1(subnet1)))
+			netSpecs = append(netSpecs, GetNetSpec(net2, GetSubnet2(subnet1)))
+			netSpecs = append(netSpecs, GetNetSpec(net3, GetSubnet3(subnet1)))
+			networks = []networkv1.IPSetNetwork{}
+			networks = append(networks, GetIPSetNet1Lower())
+			networks = append(networks, GetIPSetNet2())
+			networks = append(networks, GetIPSetNet3())
+
+			Expect(networks).To(HaveLen(len(netSpecs)))
+			// we want to ensure order is preserved so we will randomize the order of the networks
+			// and then create the IPSet
+			rand.Shuffle(len(networks), func(i, j int) {
+				networks[i], networks[j] = networks[j], networks[i]
+				netSpecs[i], netSpecs[j] = netSpecs[j], netSpecs[i]
+			})
+			// confirm that after we randomize the order the networks and netSpec are still in sync
+			for i := 0; i < len(networks); i++ {
+				Expect(networks[i].Name).To(Equal(netSpecs[i].Name))
+			}
```

Comment:
same this must not be the case in a deployment. one could have 3 networks in netSpecs (netconfig), but only 2 of them in the networks of the ipset.